test/integration/consul-container/test/gateways/http_route_test.go

```diff
@@ -256,3 +256,214 @@ func TestHTTPRouteFlattening(t *testing.T) {
 	}, checkOptions{debug: false, statusCode: service1ResponseCode, testName: "service1, v2 path with v2 hostname"})
 
 }
+
+func TestHTTPRouteParentRefChange(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+	t.Parallel()
+
+	// infrastructure set up
+	address := "localhost"
+
+	// TODO: could/should we somehow pick these randomly?
```

Comment:
I don't know if it's worth it. I think just keeping these hard-coded is fine.

Reply:
```suggestion
```